utilities/blob_db/blob_compaction_filter.cc

```diff
@@ -6,15 +6,17 @@
 #ifndef ROCKSDB_LITE
 
 #include "utilities/blob_db/blob_compaction_filter.h"
-#include "db/dbformat.h"
 
 #include <cinttypes>
 
+#include "db/dbformat.h"
+#include "test_util/sync_point.h"
+
 namespace ROCKSDB_NAMESPACE {
 namespace blob_db {
 
 BlobIndexCompactionFilterBase::~BlobIndexCompactionFilterBase() {
-  if (blob_file_) {
+  if (blob_file_ && blob_file_->GetWriter()) {
```

Comment:
The original idea was for the class to maintain the invariant `!!blob_file_ == !!writer_` (that is, either both are valid OR neither is). Instead of this change, can we fix `OpenNewBlobFileIfNeeded` so it clears both `shared_ptr`s if `CreateBlobFileAndWriter` returns failure?

Reply:
Done